security-assessment.md

```diff
@@ -0,0 +1,567 @@
+```
+
+All information, including the DFD as well as all supporting information necessary for the security assessment, should be stored in a file called `security-assessment.md` in the documentation-subfolder of each repo that is assessed.
+A template for this file can be found at [`security-assessment-template.md`](security-assessment-template.md).
+
+## Data Flow Diagram (DFD)
+
+Having a good diagram is key to a good threat model.
+
+The first step is to draw a diagram of the flow of data in your component.
+It’s the DATA flow you care about, NOT the code flow.
+Your threats come via data, NOT code.
+
+Use the following elements:
+
+<table>
+  <tr>
+    <th align="left">Element
+    <th align="left">Description
+    <th align="left">Modeling in mermaid.js
+
+  <tr>
+    <th align="left">(External) Interactor
+    <td>An element that is outside your area of control.
+    It could be a user calling into an API, it could be another component (browser, user) but not one that’s being threat modeled.
+    In general, interactors are the endpoint of the system and enter or extract data.
+    <td>
+
+    ext_interactor["External \n Interactor"]
+
+  <tr>
+    <th align="left">Process
+    <td>A process is simply some code (or microservice).
+    It does NOT mean that it’s a “process” as OS’s call processes, instead it’s just a collection of code.
+    <td>
+
+    process(("Process"))
+
+  <tr>
+    <th align="left">Data Store
+    <td>A data store is something that holds data.
+    In the DFD, there shouldn’t be any data sinks – all data gets read.
+    Also, data (usually) gets created by an interactor.
+    <td>
+
+    data_store[|borders:tb|"Data Store \n PostgreSQL"]
+
+  <tr>
+    <th align="left">Data Flow
+    <td>Represents the flow of data through the system.
+    Data always flows from an interactor via a process to a store – or vice versa.
+    <td>
+
+    process -- "Data" --> data_store
+
+  <tr>
+    <th align="left">Trust Boundary
+    <td>Occurs when one component doesn’t trust the component on the other side of the boundary.
+    There is always a trust boundary between elements running at different privilege levels.
+    <td>
+
+    subgraph "Trust Boundary"
+        process(("Process"))
+        data_store[|borders:tb|"Data Store \n PostgreSQL"]
+        process -- "Data" --> data_store
+    end
+
+  <tr>
+    <th align="left">Out of Scope Elements
+    <td>Sometimes it makes sense to add elements that are necessary for understanding the data flows but are irrelevant to the security assessments.
+    These are marked as out of scope by being grayed out and dashed.
+    <td>
+
+    ext_interactor2["External \n Interactor 2"]:::oos
+    ext_interactor2 -. "Out of scope data flow" .-> ext_interactor
+
+</table>
+
+```mermaid
+%% ----------------------
+%% -- BEST PRACTICES ----
+%% ----------------------
+%% .
+%% === OUT OF SCOPE ====
+%% mark out of scope elements with ":::oos", e.g., backendprocess:::oos

```

Comment:
space missing:
backend process

Reply:
Can’t have a space as it’s the id of that element, i guess. Not super firm with mermaid.js as of now. Will change it to `backend_process(("Backend")):::oos`